src/packagers/JSPackager.js

```diff
@@ -20,6 +21,7 @@ class JSPackager extends Packager {
   async start() {
     this.first = true;
     this.dedupe = new Map();
+    this.sourcemapPackager = new SourcemapPackager(this.bundle, this.bundler);
```

Comment:
If you register the SourcemapPackager in [here](https://github.com/parcel-bundler/parcel/blob/master/src/packagers/index.js) you won't need to construct it from here. `addAsset` will automatically get called as well, you don't need to call it from here. This way the source map packager will automatically work for non-JS langs as well if we generate maps.

Reply:
How do i go about this without overwriting the jsPackager?

Reply:
You'd register for type `map` not type `js` so it shouldn't overwrite.

```javascript
this.add('map', SourceMapPackager);
```

Parcel should already be creating a bundle of type `map` since you return one from JSAsset#generate.

Reply:
I've tried this out and it appears bundles are created based on asset.type not asset.generated.

Reply:
oh right, I changed this recently. See https://github.com/parcel-bundler/parcel/blob/master/src/Bundler.js#L413-L420. Looks like your branch doesn't have that change.